### ISSUE TITLE
Seems this code can not work for numpy array slicing

### ISSUE THREAD

**Issue by MySlientWind** [hhc97/flatliner-src#4]
Input
```python
import numpy as np
X = np.random.normal(size=(10, 10))
print(X[:, :2]**2)
```

The flatten code
```python
(lambda np: (lambda X: print((X[(::, :2:)] ** 2)))(np.random.normal(size=(10, 10))))(__import__('numpy'))
```

When run the code it gives
```python
File "<stdin>", line 1
    (lambda np: (lambda X: print((X[(::, :2:)] ** 2)))(np.random.normal(size=(10, 10))))(__import__('numpy'))
                                     ^
SyntaxError: invalid syntax
```

Maybe the flatten code should be 
```python
(lambda np: (lambda X: print((X[::, :2:] ** 2)))(np.random.normal(size=(10, 10))))(__import__('numpy'))
```

**Comment by hhc97** [comment]
Hhm, this stems from the fact that `X[:, :2]` has a tuple inside, so the output code makes it an explicit tuple which is not desired in this case.

Do you know if there are examples of slicing in python which also follow this convention, or is it a numpy specific operation?

**Comment by MySlientWind** [comment]
Though usually used in numpy related package, it's not a numpy specific operation and actually depends on `__getitem__` method. For example

```python
class demo:
    def __init__(self):
        self.array = [[1, 2, 3], [4, 5, 6], [7, 8, 9]]
        self.size = (3, 3)
    
    def __getitem__(self, key):
        x, y = key
        x = x.indices(self.size[0])
        y = y.indices(self.size[1])
        return [[self.array[xi][yi] for yi in range(*y)] for xi in range(*x)]


if __name__ == "__main__":
    d = demo()
    print(d[:, :2]) # [[1, 2], [4, 5], [7, 8]]
    print(d[:2, :]) # [[1, 2, 3], [4, 5, 6]]
```

**Comment by hhc97** [comment]
A simple fix to this might be to just get rid of any surrounding `(`'s. for the demo code you posted, that would result in this code:
```python
(lambda demo: ((lambda d: [print(d[:, :2]), print(d[:2, :])][-1])(demo()) if (__name__ == '__main__') else None))(type("demo", (), {'__init__': lambda self: [[[None for self.size in [(3, 3)]][0] for self.array in [[[1, 2, 3], [4, 5, 6], [7, 8, 9]]]][0], None][-1], '__getitem__': lambda self, key: [(lambda x: (lambda y: [[self.array[xi][yi] for yi in range(*y)] for xi in range(*x)])(y.indices(self.size[1])))(x.indices(self.size[0])) for (x, y) in [key]][0]}))
```

Which seems to work as expected:
```python
>>> (lambda demo: ((lambda d: [print(d[:, :2]), print(d[:2, :])][-1])(demo()) if (__name__ == '__main__') 
else None))(type("demo", (), {'__init__': lambda self: [[[None for self.size in [(3, 3)]][0] for self.arra
y in [[[1, 2, 3], [4, 5, 6], [7, 8, 9]]]][0], None][-1], '__getitem__': lambda self, key: [(lambda x: (lam
bda y: [[self.array[xi][yi] for yi in range(*y)] for xi in range(*x)])(y.indices(self.size[1])))(x.indices
(self.size[0])) for (x, y) in [key]][0]}))
[[1, 2], [4, 5], [7, 8]]
[[1, 2, 3], [4, 5, 6]]
```

Before I push this out though, I'm wondering if there are any other cases of weird `__getitem__` implementations that this would break, what do you think?

**Comment by MySlientWind** [comment]
That's all WEIRD `__getitem__` implementations as I know
```python
class demo0:
    def __init__(self):
        self.array = [[1, 2, 3], [4, 5, 6], [7, 8, 9]]
        self.size = (3, 3)
    
    def __getitem__(self, key):
        x, y = key
        x = x.indices(self.size[0])
        y = y.indices(self.size[1])
        return [[self.array[xi][yi] for yi in range(*y)] for xi in range(*x)]

class demo1:
    def __init__(self):
        self.array = [[1, 2, 3], [4, 5, 6], [7, 8, 9]]
    
    def __getitem__(self, key):
        x, y = key
        return self.array[x][y]
        

class demo2:
    def __init__(self):
        self.dict = {'a': [1, 2, 3], 'b': [4, 5, 6], 'c': [7, 8, 9]}
    
    def __getitem__(self, key):
        ks, x = key
        return {k : self.dict[k][x] for k in ks}

if __name__ == "__main__":
    d0 = demo0()
    print(d0[:, :2]) # [[1, 2], [4, 5], [7, 8]]
    print(d0[:2, :]) # [[1, 2, 3], [4, 5, 6]]

    d1 = demo1()
    print(d1[(1, 2)]) # 6

    d2 = demo2()
    print(d2[('a', 'b'), :2]) # {'a': [1, 2], 'b': [4, 5]}
```

**Comment by hhc97** [comment]
I suppose the issue is that python doesn't allow a slice to be in a tuple, `(::, ::)` will give an error, while `::, ::` is fine if its in a subscript

**Comment by hhc97** [comment]
Alright, fixed in #6, thank you very much for the examples you provided, the changes have been deployed as well, see https://flatliner.herokuapp.com/3f096006ef413f0d0668ce14579471d012e1768c and https://flatliner.herokuapp.com/ad7e18230369cd7bc1528adb472dc121e3d95feb

Let me know if you notice anything else that doesn't seem quite right!
